src/axom/primal/tests/primal_rational_bezier.cpp

```diff
@@ -0,0 +1,394 @@
+    PointType {1.8125, 1.85, 1.8875},
+    PointType {2.365, 2.32, 2.225}};
+  double b3Weights[order + 1] = {1.0, 1.5, 2.0, 2.5};
+
+  PointType b4Nodes[order + 1] = {PointType {2.365, 2.32, 2.225},
+                                  PointType {41.0 / 15.0, 79 / 30.0, 2.45},
+                                  PointType {43.0 / 14.0, 106.0 / 35.0, 2.7},
+                                  PointType {3.2, 3.5, 3.0}};
+  double b4Weights[order + 1] = {2.5, 3.0, 3.5, 4.0};
+
+  BezierCurveType b3True(b3Nodes, b3Weights, 3);
+  BezierCurveType b4True(b4Nodes, b4Weights, 3);
+
+  for(int i = 0; i < DIM; ++i)
+  {
+    for(int p = 0; p <= order; ++p)
+    {
+      EXPECT_NEAR(b3Curve[p][i], b3True[p][i], 1e-15);
+      EXPECT_NEAR(b4Curve[p][i], b4True[p][i], 1e-15);
+    }
+  }
+}
+
+//------------------------------------------------------------------------------
+TEST(primal_rationalbezier, split_point)
+{
+  SLIC_INFO("Testing rational Bezier splitting for order 0");
+
+  const int DIM = 2;
+  using CoordType = double;
+  using PointType = primal::Point<CoordType, DIM>;
+  using BezierCurveType = primal::BezierCurve<CoordType, DIM>;
+
+  // Test order-0
+  {
+    const int order = 0;
+    PointType data[order + 1] = {PointType::make_point(0.6, 1.2)};
+    double weights[order + 1] = {1.5};
+
+    BezierCurveType b(data, weights, order);
+
+    BezierCurveType c1, c2;
+    b.split(0.5, c1, c2);
+
+    EXPECT_NEAR(weights[0], c1.getWeight(0), 1e-15);
+    EXPECT_NEAR(weights[0], c2.getWeight(0), 1e-15);
+
+    for(int i = 0; i < DIM; ++i)
+    {
+      EXPECT_NEAR(data[0][i], c1[0][i], 1e-15);
+      EXPECT_NEAR(data[0][i], c2[0][i], 1e-15);
+    }
+  }
+}
+
+TEST(primal_rationalbezier, split_linear)
+{
+  SLIC_INFO("Testing rational Bezier splitting for order 1");
+
+  const int DIM = 2;
+  using CoordType = double;
+  using PointType = primal::Point<CoordType, DIM>;
+  using BezierCurveType = primal::BezierCurve<CoordType, DIM>;
+
+  const int order = 1;
+  PointType data[order + 1] = {PointType {-1, -5}, PointType {1, 5}};
+  double weights[order + 1] = {2, 8};
+
+  BezierCurveType b(data, weights, order);
+
+  {
+    BezierCurveType c1, c2;
+    b.split(0.5, c1, c2);
+
+    EXPECT_DOUBLE_EQ(-1., c1[0][0]);
+    EXPECT_DOUBLE_EQ(-5., c1[0][1]);
+    EXPECT_DOUBLE_EQ(0.6, c1[1][0]);
+    EXPECT_DOUBLE_EQ(3.0, c1[1][1]);
+
+    EXPECT_DOUBLE_EQ(0.6, c2[0][0]);
+    EXPECT_DOUBLE_EQ(3.0, c2[0][1]);
+    EXPECT_DOUBLE_EQ(1.0, c2[1][0]);
+    EXPECT_DOUBLE_EQ(5.0, c2[1][1]);
+  }
+
+  {
+    BezierCurveType c1, c2;
+    b.split(0.2, c1, c2);
+
+    for(int i = 0; i < DIM; ++i)
+    {
+      EXPECT_DOUBLE_EQ(-1., c1[0][0]);
+      EXPECT_DOUBLE_EQ(-5., c1[0][1]);
+      EXPECT_DOUBLE_EQ(0.0, c1[1][0]);
+      EXPECT_DOUBLE_EQ(0.0, c1[1][1]);
+
+      EXPECT_DOUBLE_EQ(0.0, c2[0][0]);
+      EXPECT_DOUBLE_EQ(0.0, c2[0][1]);
+      EXPECT_DOUBLE_EQ(1.0, c2[1][0]);
+      EXPECT_DOUBLE_EQ(5.0, c2[1][1]);
+    }
```

Comment:
These lines caused some test failures on LLNL's blueos host-configs:

```
145: [INFO] Testing rational Bezier splitting for order 1 
145: ///axom/src/axom/primal/tests/primal_rational_bezier.cpp:268: Failure
145: Expected equality of these values:
145:   0.0
145:     Which is: 0
145:   c1[1][1]
145:     Which is: -1.3877787807814457e-16
145: ///axom/src/axom/primal/tests/primal_rational_bezier.cpp:271: Failure
145: Expected equality of these values:
145:   0.0
145:     Which is: 0
145:   c2[0][1]
145:     Which is: -1.3877787807814457e-16
145: ///axom/src/axom/primal/tests/primal_rational_bezier.cpp:268: Failure
145: Expected equality of these values:
145:   0.0
145:     Which is: 0
145:   c1[1][1]
145:     Which is: -1.3877787807814457e-16
145: ///axom/src/axom/primal/tests/primal_rational_bezier.cpp:271: Failure
145: Expected equality of these values:
145:   0.0
145:     Which is: 0
145:   c2[0][1]
145:     Which is: -1.3877787807814457e-16
145: [  FAILED  ] primal_rationalbezier.split_linear (0 ms)
```

Might be resolved using `EXPECT_NEAR` with an exlicit `EPS` around 1e-12 ?